### PR TITLE
Update field pattern to recognize type aliases.

### PR DESCRIPTION
**Pull request by thomasvandoren**
Previously, the pattern would not match something like: `type age = int(64)`.

Also, include tests to exercise the regression.

### CODE REVIEW

**Comment by thomasvandoren** [comment]
@lydia-duncan -- will you please look at this, post merge?

File: test/test_chapeldomain.py [hunk 1]
```diff
@@ -586,6 +586,9 @@ def test_with_all(self):
             ('var RandomStreamPrivate_lock$: _syncvar(bool)', 'var ', None, 'RandomStreamPrivate_lock$', ': _syncvar(bool)'),
             ('var RandomStreamPrivate_lock$: sync bool', 'var ', None, 'RandomStreamPrivate_lock$', ': sync bool'),
             ('const RS$.lock$: sync MyMod$.MyClass$.bool', 'const ', 'RS$.', 'lock$', ': sync MyMod$.MyClass$.bool'),
+            ('type commDiagnostics = chpl_commDiagnostics', 'type ', None, 'commDiagnostics', ' = chpl_commDiagnostics'),
+            ('type age = int(64)', 'type ', None, 'age', ' = int(64)'),
+            ('type MyMod.BigAge=BigNum.BigInt', 'type ', 'MyMod.', 'BigAge', '=BigNum.BigInt'),
```
Comment (lydia-duncan) [inline test/test_chapeldomain.py:591]: I'd like a case with both a defined type and a default definition included (and maybe one with a just default definition that isn't used with a type - I could be missing that case, though), but other than that, this looks great!

Comment (thomasvandoren) [inline test/test_chapeldomain.py:591]: I'm not sure what you're asking... Do you want tests for something like:

``` chapel
type blah: <something??> = int(64)
type blah: <something??>
```

Comment (lydia-duncan) [inline test/test_chapeldomain.py:591]: More like:

var blah: int(64) = 4
const blah = 15

You may have those already, but I didn't see them when I scrolled in this file.

Comment (thomasvandoren) [inline test/test_chapeldomain.py:591]: Oh, I see. Yes, I'll add cases for those.

**Pull request merged**